<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Coby Warkentin and Donghyung Lee" />

<meta name="date" content="2023-06-23" />

<title>Evaluating KOMPUTE - Body Composition Data</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">komputeExamples</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statsleelab/komputeExamples">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Evaluating KOMPUTE - Body Composition
Data</h1>
<h4 class="author">Coby Warkentin and Donghyung Lee</h4>
<h4 class="date">2023-06-23</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#import-necessary-packages"
id="toc-import-necessary-packages">Import necessary packages</a></li>
<li><a href="#preparing-control-phenotype-data"
id="toc-preparing-control-phenotype-data">Preparing control phenotype
data</a>
<ul>
<li><a href="#importing-body-composition-control-phenotype-dataset"
id="toc-importing-body-composition-control-phenotype-dataset">Importing
Body Composition Control Phenotype Dataset</a></li>
<li><a href="#visualizing-measured-phenotypes-via-a-heatmap"
id="toc-visualizing-measured-phenotypes-via-a-heatmap">Visualizing
measured phenotypes via a heatmap</a></li>
<li><a href="#exclude-phenotypes-with-fewer-than-15000-observations"
id="toc-exclude-phenotypes-with-fewer-than-15000-observations">Exclude
phenotypes with fewer than 15,000 observations</a></li>
<li><a href="#remove-samples-with-fewer-than-7-measured-phenotypes"
id="toc-remove-samples-with-fewer-than-7-measured-phenotypes">Remove
samples with fewer than 7 measured phenotypes</a></li>
<li><a href="#heapmap-of-filtered-phenotypes"
id="toc-heapmap-of-filtered-phenotypes">Heapmap of filtered
phenotypes</a></li>
<li><a href="#reforatting-the-dataset-long-to-wide"
id="toc-reforatting-the-dataset-long-to-wide">Reforatting the dataset
(long to wide)</a></li>
<li><a href="#visualizing-phenotype-distributions"
id="toc-visualizing-phenotype-distributions">Visualizing phenotype
distributions</a></li>
<li><a href="#rank-z-transformation" id="toc-rank-z-transformation">Rank
Z transformation</a></li>
<li><a href="#conducting-principal-variance-component-analysis-pvca"
id="toc-conducting-principal-variance-component-analysis-pvca">Conducting
Principal Variance Component Analysis (PVCA)</a></li>
<li><a href="#batch-effect-removal-using-combat"
id="toc-batch-effect-removal-using-combat">Batch effect removal using
ComBat</a></li>
<li><a href="#pvca-on-combat-residuals"
id="toc-pvca-on-combat-residuals">PVCA on ComBat residuals</a></li>
<li><a href="#supplementary-figure"
id="toc-supplementary-figure">Supplementary figure</a></li>
<li><a href="#computing-phenotypic-correlations"
id="toc-computing-phenotypic-correlations">Computing phenotypic
correlations</a></li>
</ul></li>
<li><a href="#preparation-of-impc-summary-statistics-data"
id="toc-preparation-of-impc-summary-statistics-data">Preparation of IMPC
summary statistics data</a>
<ul>
<li><a href="#loading-body-composition-summary-stat-impcv16"
id="toc-loading-body-composition-summary-stat-impcv16">Loading Body
Composition summary stat (IMPCv16)</a></li>
<li><a href="#visualizing-gene-phenotype-pair-duplicates"
id="toc-visualizing-gene-phenotype-pair-duplicates">Visualizing
gene-phenotype pair duplicates</a></li>
<li><a
href="#consolidating-muliple-z-scores-of-a-gene-phenotype-pair-using-stouffers-method"
id="toc-consolidating-muliple-z-scores-of-a-gene-phenotype-pair-using-stouffers-method">Consolidating
muliple z-scores of a gene-phenotype pair using Stouffer’s
Method</a></li>
<li><a href="#generating-z-score-matrix-reformatting"
id="toc-generating-z-score-matrix-reformatting">Generating Z-score
matrix (reformatting)</a></li>
<li><a href="#visualization-of-phenotype-gene-coverage"
id="toc-visualization-of-phenotype-gene-coverage">Visualization of
Phenotype-Gene Coverage</a></li>
<li><a href="#distribution-of-z-scores-across-phenotypes"
id="toc-distribution-of-z-scores-across-phenotypes">Distribution of
Z-Scores Across Phenotypes</a></li>
<li><a href="#estimation-of-genetic-correlation-matrix-using-z-scores"
id="toc-estimation-of-genetic-correlation-matrix-using-z-scores">Estimation
of Genetic Correlation Matrix Using Z-Scores</a></li>
<li><a
href="#comparison-of-phenotypic-correlation-and-genetic-correlation-among-phenotypes"
id="toc-comparison-of-phenotypic-correlation-and-genetic-correlation-among-phenotypes">Comparison
of Phenotypic Correlation and Genetic Correlation Among
Phenotypes</a></li>
<li><a
href="#correlation-analysis-between-genetic-correlation-matrices-using-mantels-test"
id="toc-correlation-analysis-between-genetic-correlation-matrices-using-mantels-test">Correlation
Analysis Between Genetic Correlation Matrices Using Mantel’s
Test</a></li>
</ul></li>
<li><a href="#evaluating-the-kompute-imputation-algorithm"
id="toc-evaluating-the-kompute-imputation-algorithm">Evaluating the
KOMPUTE Imputation Algorithm</a>
<ul>
<li><a href="#initializing-the-kompute-package"
id="toc-initializing-the-kompute-package">Initializing the KOMPUTE
Package</a></li>
<li><a
href="#simulation-study---comparison-of-imputed-vs-measured-z-score-values"
id="toc-simulation-study---comparison-of-imputed-vs-measured-z-score-values">Simulation
study - Comparison of imputed vs measured z-score values</a></li>
<li><a href="#run-kompute-method" id="toc-run-kompute-method">Run
KOMPUTE method</a></li>
<li><a href="#save-z-score-matrix-and-phenotype-correlation-matrix"
id="toc-save-z-score-matrix-and-phenotype-correlation-matrix">Save
z-score matrix and phenotype correlation matrix</a></li>
</ul></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>komputeExamples/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230110code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230110)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230110code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230110)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstatsleelabkomputeExamplestreefbd0f5c6d44a523201f2a46c03e59b12210b3a71targetblankfbd0f5ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/statsleelab/komputeExamples/tree/fbd0f5c6d44a523201f2a46c03e59b12210b3a71" target="_blank">fbd0f5c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstatsleelabkomputeExamplestreefbd0f5c6d44a523201f2a46c03e59b12210b3a71targetblankfbd0f5ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/statsleelab/komputeExamples/tree/fbd0f5c6d44a523201f2a46c03e59b12210b3a71" target="_blank">fbd0f5c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/kompute_test_BC_v16.Rmd</code>)
and HTML (<code>docs/kompute_test_BC_v16.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/fbd0f5c6d44a523201f2a46c03e59b12210b3a71/analysis/kompute_test_BC_v16.Rmd" target="_blank">fbd0f5c</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-23
</td>
<td>
kompute input format changed
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/statsleelab/komputeExamples/e7a365f2b92c6ab6a86d596b13e93908c8534fac/docs/kompute_test_BC_v16.html" target="_blank">e7a365f</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-21
</td>
<td>
pheno.cor added
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/fd859e3c29d961d8c0f16a6cdda58a8d18f89b2f/analysis/kompute_test_BC_v16.Rmd" target="_blank">fd859e3</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-21
</td>
<td>
BC pheno cor added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/statsleelab/komputeExamples/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/kompute_test_BC_v16.html" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
<td>
doc updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/d343f6aaf3961a6880ad1f6153e73e3a534d04c4/analysis/kompute_test_BC_v16.Rmd" target="_blank">d343f6a</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
<td>
doc updated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/statsleelab/komputeExamples/d56cc7aaadf25c2dbd667b52b8458b78fae74c49/docs/kompute_test_BC_v16.html" target="_blank">d56cc7a</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/statsleelab/komputeExamples/989feb60f6281fb4eaf2cfcfce252144832ee843/docs/kompute_test_BC_v16.html" target="_blank">989feb6</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
<td>
v16 update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/f4e1ab96962e4e3061367a49df521b7dd22ec014/analysis/kompute_test_BC_v16.Rmd" target="_blank">f4e1ab9</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
<td>
v16 update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/statsleelab/komputeExamples/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/kompute_test_BC_v16.html" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
<td>
first commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/32dd775bf5907abd2ae88a2a60a3f9cb859c9d40/analysis/kompute_test_BC_v16.Rmd" target="_blank">32dd775</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
<td>
first commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="import-necessary-packages" class="section level2">
<h2>Import necessary packages</h2>
<pre class="r"><code>rm(list=ls())
knitr::opts_chunk$set(message = FALSE, warning = FALSE)

library(data.table)
library(dplyr)
library(reshape2)
library(ggplot2)
library(tidyr) #spread
library(RColorBrewer)
library(circlize)
library(ComplexHeatmap)
library(patchwork)
library(ggpubr)</code></pre>
</div>
<div id="preparing-control-phenotype-data" class="section level2">
<h2>Preparing control phenotype data</h2>
<div id="importing-body-composition-control-phenotype-dataset"
class="section level3">
<h3>Importing Body Composition Control Phenotype Dataset</h3>
<pre class="r"><code>BC.data &lt;- readRDS(&quot;data/BC.data.rds&quot;)
dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
</div>
<div id="visualizing-measured-phenotypes-via-a-heatmap"
class="section level3">
<h3>Visualizing measured phenotypes via a heatmap</h3>
<p>The heatmap below presents a visualization of the phenotypic
measurements taken for each control mouse. The columns represent
individual mice, while the rows correspond to the distinct phenotypes
measured.</p>
<pre class="r"><code>mtest &lt;- table(BC.data$proc_param_name_stable_id, BC.data$biological_sample_id)
mtest &lt;-as.data.frame.matrix(mtest)
dim(mtest)</code></pre>
<pre><code>[1]    51 25690</code></pre>
<pre class="r"><code>if(FALSE){
nmax &lt;-max(mtest)
library(circlize)
col_fun = colorRamp2(c(0, nmax), c(&quot;white&quot;, &quot;red&quot;))
col_fun(seq(0, nmax))
ht = Heatmap(as.matrix(mtest), cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, col = col_fun,
             row_names_gp = gpar(fontsize = 8), name=&quot;Count&quot;)
draw(ht)
}</code></pre>
</div>
<div id="exclude-phenotypes-with-fewer-than-15000-observations"
class="section level3">
<h3>Exclude phenotypes with fewer than 15,000 observations</h3>
<p>To maintain data quality and robustness, we will discard any
phenotypes that have fewer than 15,000 recorded observations.</p>
<pre class="r"><code>mtest &lt;- table(BC.data$proc_param_name, BC.data$biological_sample_id)
dim(mtest)</code></pre>
<pre><code>[1]    10 25690</code></pre>
<pre class="r"><code>#head(mtest[,1:10])
mtest0 &lt;- mtest&gt;0
#head(mtest0[,1:10])
rowSums(mtest0)</code></pre>
<pre><code>                       BC_BMC/Body weight 
                                    24446 
                           BC_Body length 
                                    19233 
                           BC_Body weight 
                                    25675 
                             BC_Bone Area 
                                    24461 
BC_Bone Mineral Content (excluding skull) 
                                    24714 
BC_Bone Mineral Density (excluding skull) 
                                    24715 
                              BC_Fat mass 
                                    24950 
                       BC_Fat/Body weight 
                                    24682 
                             BC_Lean mass 
                                    24952 
                      BC_Lean/Body weight 
                                    24684 </code></pre>
<pre class="r"><code>rmv.pheno.list &lt;- rownames(mtest)[rowSums(mtest0)&lt;15000]
#rmv.pheno.list
dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
<pre class="r"><code>BC.data &lt;- BC.data %&gt;% filter(!(proc_param_name %in% rmv.pheno.list))
dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
<pre class="r"><code># number of phenotypes left
length(unique(BC.data$proc_param_name))</code></pre>
<pre><code>[1] 10</code></pre>
</div>
<div id="remove-samples-with-fewer-than-7-measured-phenotypes"
class="section level3">
<h3>Remove samples with fewer than 7 measured phenotypes</h3>
<pre class="r"><code>mtest &lt;- table(BC.data$proc_param_name, BC.data$biological_sample_id)
dim(mtest)</code></pre>
<pre><code>[1]    10 25690</code></pre>
<pre class="r"><code>head(mtest[,1:10])</code></pre>
<pre><code>                                           
                                            39638 39639 39640 39641 39642 39643
  BC_BMC/Body weight                            1     1     1     1     1     1
  BC_Body length                                0     0     0     1     0     1
  BC_Body weight                                2     1     1     1     1     1
  BC_Bone Area                                  1     1     1     1     1     1
  BC_Bone Mineral Content (excluding skull)     2     1     1     1     1     1
  BC_Bone Mineral Density (excluding skull)     2     1     1     1     1     1
                                           
                                            39650 39651 39652 39657
  BC_BMC/Body weight                            1     1     1     1
  BC_Body length                                1     0     0     1
  BC_Body weight                                1     1     1     1
  BC_Bone Area                                  1     1     1     1
  BC_Bone Mineral Content (excluding skull)     1     1     1     1
  BC_Bone Mineral Density (excluding skull)     1     1     1     1</code></pre>
<pre class="r"><code>mtest0 &lt;- mtest&gt;0
head(mtest0[,1:10])</code></pre>
<pre><code>                                           
                                            39638 39639 39640 39641 39642 39643
  BC_BMC/Body weight                         TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Body length                            FALSE FALSE FALSE  TRUE FALSE  TRUE
  BC_Body weight                             TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Bone Area                               TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Content (excluding skull)  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Density (excluding skull)  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
                                           
                                            39650 39651 39652 39657
  BC_BMC/Body weight                         TRUE  TRUE  TRUE  TRUE
  BC_Body length                             TRUE FALSE FALSE  TRUE
  BC_Body weight                             TRUE  TRUE  TRUE  TRUE
  BC_Bone Area                               TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Content (excluding skull)  TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Density (excluding skull)  TRUE  TRUE  TRUE  TRUE</code></pre>
<pre class="r"><code>summary(colSums(mtest0))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    9.00   10.00    9.44   10.00   10.00 </code></pre>
<pre class="r"><code>rmv.sample.list &lt;- colnames(mtest)[colSums(mtest0)&lt;7]
length(rmv.sample.list)</code></pre>
<pre><code>[1] 1829</code></pre>
<pre class="r"><code>dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
<pre class="r"><code>BC.data &lt;- BC.data %&gt;% filter(!(biological_sample_id %in% rmv.sample.list))
dim(BC.data)</code></pre>
<pre><code>[1] 234305     10</code></pre>
<pre class="r"><code># number of observations to use
length(unique(BC.data$biological_sample_id))</code></pre>
<pre><code>[1] 23861</code></pre>
</div>
<div id="heapmap-of-filtered-phenotypes" class="section level3">
<h3>Heapmap of filtered phenotypes</h3>
<pre class="r"><code>if(FALSE){
mtest &lt;- table(BC.data$proc_param_name, BC.data$biological_sample_id)
dim(mtest)
mtest &lt;-as.data.frame.matrix(mtest)
nmax &lt;-max(mtest)
library(circlize)
col_fun = colorRamp2(c(0, nmax), c(&quot;white&quot;, &quot;red&quot;))
col_fun(seq(0, nmax))
pdf(&quot;~/Google Drive Miami/Miami_IMPC/output/measured_phenotypes_controls_after_filtering_BC.pdf&quot;, width = 10, height = 3)
ht = Heatmap(as.matrix(mtest), cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, col = col_fun,
             row_names_gp = gpar(fontsize = 7), name=&quot;Count&quot;)
draw(ht)
dev.off()
}</code></pre>
</div>
<div id="reforatting-the-dataset-long-to-wide" class="section level3">
<h3>Reforatting the dataset (long to wide)</h3>
<p>We restructure our data from a long format to a wide one for further
processing and analysis.</p>
<pre class="r"><code>BC.mat &lt;- BC.data %&gt;% 
  dplyr::select(biological_sample_id, proc_param_name, data_point, sex, phenotyping_center, strain_name) %&gt;% 
  ##consider weight or age in weeks
  arrange(biological_sample_id) %&gt;%
  distinct(biological_sample_id, proc_param_name, .keep_all=TRUE) %&gt;% ## remove duplicates, maybe mean() is better.
  spread(proc_param_name, data_point) %&gt;%
  tibble::column_to_rownames(var=&quot;biological_sample_id&quot;)
head(BC.mat)</code></pre>
<pre><code>         sex phenotyping_center strain_name BC_BMC/Body weight BC_Body length
39638 female        MRC Harwell C57BL/6NTac          0.0567631             NA
39639 female               HMGU C57BL/6NCrl          0.0224897             NA
39640 female               HMGU C57BL/6NTac          0.0214276             NA
39641   male               HMGU C57BL/6NCrl          0.0191929            9.7
39642 female        MRC Harwell C57BL/6NTac          0.0242145             NA
39643 female               HMGU C57BL/6NCrl          0.0224004            9.4
      BC_Body weight BC_Bone Area BC_Bone Mineral Content (excluding skull)
39638          22.12     13.51560                                   1.25560
39639          23.30      8.11161                                   0.52401
39640          23.20      6.85683                                   0.49712
39641          29.70      8.61072                                   0.57003
39642          25.27      8.42837                                   0.61190
39643          24.30      8.42616                                   0.54433
      BC_Bone Mineral Density (excluding skull) BC_Fat mass BC_Fat/Body weight
39638                                    0.0929      6.9362           0.313571
39639                                    0.0646      6.1000           0.261803
39640                                    0.0725      3.3000           0.142241
39641                                    0.0662      7.1000           0.239057
39642                                    0.0726      3.4382           0.136059
39643                                    0.0646      6.9000           0.283951
      BC_Lean mass BC_Lean/Body weight
39638      11.2569            0.508901
39639      14.7000            0.630901
39640      17.1000            0.737069
39641      21.7000            0.730640
39642      20.5392            0.812790
39643      15.3000            0.629630</code></pre>
<pre class="r"><code>dim(BC.mat)</code></pre>
<pre><code>[1] 23861    13</code></pre>
<pre class="r"><code>summary(colSums(is.na(BC.mat[,-1:-3])))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.00    0.00    0.00  532.50    0.75 5323.00 </code></pre>
</div>
<div id="visualizing-phenotype-distributions" class="section level3">
<h3>Visualizing phenotype distributions</h3>
<pre class="r"><code>ggplot(melt(BC.mat), aes(x=value)) + 
  geom_histogram() + 
  facet_wrap(~variable, scales=&quot;free&quot;, ncol=5)+
  theme(strip.text.x = element_text(size = 6))</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/pheno_dist-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pheno_dist-1">
Past versions of pheno_dist-1.png
</button>
</p>
<div id="fig-pheno_dist-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/pheno_dist-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="rank-z-transformation" class="section level3">
<h3>Rank Z transformation</h3>
<p>In this step, we conduct a rank Z transformation on the phenotype
data to ensure that the data is normally distributed</p>
<pre class="r"><code>library(RNOmni)
BC.mat.rank &lt;- BC.mat
dim(BC.mat.rank)</code></pre>
<pre><code>[1] 23861    13</code></pre>
<pre class="r"><code>BC.mat.rank &lt;- BC.mat.rank[complete.cases(BC.mat.rank),]
dim(BC.mat.rank)</code></pre>
<pre><code>[1] 18538    13</code></pre>
<pre class="r"><code>dim(BC.mat)</code></pre>
<pre><code>[1] 23861    13</code></pre>
<pre class="r"><code>BC.mat &lt;- BC.mat[complete.cases(BC.mat),]
dim(BC.mat)</code></pre>
<pre><code>[1] 18538    13</code></pre>
<pre class="r"><code>BC.mat.rank &lt;- cbind(BC.mat.rank[,1:3], apply(BC.mat.rank[,-1:-3], 2, RankNorm))
ggplot(melt(BC.mat.rank), aes(x=value)) + 
  geom_histogram() + 
  facet_wrap(~variable, scales=&quot;free&quot;, ncol=5)+
  theme(strip.text.x = element_text(size = 6))</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/rankZ_trans-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rankZ_trans-1">
Past versions of rankZ_trans-1.png
</button>
</p>
<div id="fig-rankZ_trans-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/rankZ_trans-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="conducting-principal-variance-component-analysis-pvca"
class="section level3">
<h3>Conducting Principal Variance Component Analysis (PVCA)</h3>
<p>In this step, we apply Principal Variance Component Analysis (PVCA)
on the phenotype matrix data. PVCA is an approach that combines
Principal Component Analysis (PCA) and Variance Component Analysis to
quantify the proportion of total variance in the data attributed to each
important covariate, in this case ‘sex’ and ‘phenotyping_center’.</p>
<p>First, we prepare our metadata which includes our chosen covariates.
Any character variables in the metadata are then converted to factors.
To avoid potential confounding, we check for associations between our
covariates and drop ‘strain_name’ due to its strong association with
‘phenotyping_center’.</p>
<p>Next, we run PVCA on randomly chosen subsets of our phenotype data
(for computational efficiency). Finally, we compute the average effect
size across all random samples and visualize the results in a PVCA
plot.</p>
<pre class="r"><code>source(&quot;code/PVCA.R&quot;)

meta &lt;- BC.mat.rank[,1:3] ## examining covariates sex, phenotyping_center, and strain_name
head(meta)</code></pre>
<pre><code>         sex phenotyping_center strain_name
39641   male               HMGU C57BL/6NCrl
39643 female               HMGU C57BL/6NCrl
39657   male               HMGU C57BL/6NCrl
39750 female               HMGU C57BL/6NCrl
39763 female               HMGU C57BL/6NCrl
39773 female               HMGU C57BL/6NCrl</code></pre>
<pre class="r"><code>dim(meta)</code></pre>
<pre><code>[1] 18538     3</code></pre>
<pre class="r"><code>summary(meta) # variables are currently characters</code></pre>
<pre><code>     sex            phenotyping_center strain_name       
 Length:18538       Length:18538       Length:18538      
 Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character  </code></pre>
<pre class="r"><code>meta[sapply(meta, is.character)] &lt;- lapply(meta[sapply(meta, is.character)], as.factor)
summary(meta) # now all variables are converted to factors</code></pre>
<pre><code>     sex       phenotyping_center                     strain_name  
 female:9282   WTSI    :4573      B6Brd;B6Dnk;B6N-Tyr&lt;c-Brd&gt;: 176  
 male  :9256   JAX     :3675      C57BL/6N                  :9127  
               ICS     :2075      C57BL/6NCrl               :2997  
               BCM     :1832      C57BL/6NJ                 :3675  
               RBRC    :1692      C57BL/6NJcl               : 728  
               UC Davis:1441      C57BL/6NTac               :1835  
               (Other) :3250                                       </code></pre>
<pre class="r"><code>chisq.test(meta[,1],meta[,2])</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  meta[, 1] and meta[, 2]
X-squared = 13.572, df = 10, p-value = 0.1934</code></pre>
<pre class="r"><code>chisq.test(meta[,2],meta[,3]) </code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  meta[, 2] and meta[, 3]
X-squared = 59770, df = 50, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>meta&lt;-meta[,-3] # phenotyping_center and strain_name strongly associated which could cause confounding in the PVCA analysis, so we drop &#39;strain_name&#39;.

G &lt;- t(BC.mat.rank[,-1:-3]) ## preparing the phenotype matrix data

set.seed(09302021)

# Perform PVCA for 10 random samples of size 1000 (more computationally efficient)
pvca.res &lt;- matrix(nrow=10, ncol=3)
for (i in 1:10){
  sample &lt;- sample(1:ncol(G), 1000, replace=FALSE)
  pvca.res[i,] &lt;- PVCA(G[,sample], meta[sample,], threshold=0.6, inter=FALSE)
}

# Compute average effect size across the 10 random samples
pvca.means &lt;- colMeans(pvca.res)
names(pvca.means) &lt;- c(colnames(meta), &quot;resid&quot;)

# Create PVCA plot
pvca.plot &lt;- PlotPVCA(pvca.means, &quot;PVCA of Phenotype Matrix Data&quot;)
pvca.plot</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/PVCA-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PVCA-1">
Past versions of PVCA-1.png
</button>
</p>
<div id="fig-PVCA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/PVCA-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(file=&quot;docs/figure/figures.Rmd/pvca_BC_1_v16.png&quot;, width=600, height=350)
pvca.plot
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="batch-effect-removal-using-combat" class="section level3">
<h3>Batch effect removal using ComBat</h3>
<p>We remove batch effects (the center effect) in the phenotype data set
by using the ComBat method.</p>
<pre class="r"><code>library(sva)
combat_komp = ComBat(dat=G, batch=meta$phenotyping_center, par.prior=TRUE, prior.plots=TRUE, mod=NULL)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/combat-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-combat-1">
Past versions of combat-1.png
</button>
</p>
<div id="fig-combat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/combat-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>combat_komp[1:5,1:5]</code></pre>
<pre><code>                                               39641      39643      39657
BC_BMC/Body weight                        -0.5446577  0.7624261 -0.8478037
BC_Body length                             0.2371464 -0.1563519  0.2371464
BC_Body weight                             0.9069478 -0.3865968  0.9844317
BC_Bone Area                               0.4188173  0.1510426  1.0646332
BC_Bone Mineral Content (excluding skull)  0.5872088  0.1561049  0.3622582
                                                39750       39763
BC_BMC/Body weight                        -0.06815544  0.14454428
BC_Body length                            -0.28891231 -0.01379022
BC_Body weight                            -0.59375686 -0.59375686
BC_Bone Area                              -0.51719766 -0.29607359
BC_Bone Mineral Content (excluding skull) -0.84928634 -0.64311406</code></pre>
<pre class="r"><code>G[1:5,1:5] # for comparison, combat_komp is same form and same dimensions as G</code></pre>
<pre><code>                                                39641      39643      39657
BC_BMC/Body weight                         0.61537676  1.3908581 0.43552296
BC_Body length                             0.04917004 -0.1811585 0.04917004
BC_Body weight                             0.72940618 -0.3554534 0.79438976
BC_Bone Area                              -0.05160698 -0.2731793 0.48277848
BC_Bone Mineral Content (excluding skull)  1.43981495  1.1313384 1.27885140
                                               39750       39763
BC_BMC/Body weight                         0.8980814  1.02427422
BC_Body length                            -0.2587508 -0.09771206
BC_Body weight                            -0.5291927 -0.52919275
BC_Bone Area                              -0.8261201 -0.64314918
BC_Bone Mineral Content (excluding skull)  0.4119303  0.55945691</code></pre>
</div>
<div id="pvca-on-combat-residuals" class="section level3">
<h3>PVCA on ComBat residuals</h3>
<p>After using ComBat to account for batch effects, we perform a PVCA on
the residuals. We expect to observe a significantly reduced effect from
the phenotyping centers.</p>
<pre class="r"><code>set.seed(09302021)
# Perform PVCA for 10 random samples (more computationally efficient)
pvca.res.nobatch &lt;- matrix(nrow=10, ncol=3)
for (i in 1:10){
  sample &lt;- sample(1:ncol(combat_komp), 1000, replace=FALSE)
  pvca.res.nobatch[i,] &lt;- PVCA(combat_komp[,sample], meta[sample,], threshold=0.6, inter=FALSE)
}

# Compute average effect size across samples
pvca.means.nobatch &lt;- colMeans(pvca.res.nobatch)
names(pvca.means.nobatch) &lt;- c(colnames(meta), &quot;resid&quot;)

# Generate PVCA plot
pvca.plot.nobatch &lt;- PlotPVCA(pvca.means.nobatch, &quot;PVCA of Phenotype Matrix Data with Reduced Batch Effect&quot;)
pvca.plot.nobatch</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/PVCA_resid-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-PVCA_resid-1">
Past versions of PVCA_resid-1.png
</button>
</p>
<div id="fig-PVCA_resid-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/PVCA_resid-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(file=&quot;docs/figure/figures.Rmd/pvca_BC_2_v16.png&quot;, width=600, height=350)
pvca.plot.nobatch
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="supplementary-figure" class="section level3">
<h3>Supplementary figure</h3>
<pre class="r"><code># Supplementary Figure 1
pvca.plot &lt;- pvca.plot +
  labs(title=&quot;&quot;) +
  scale_x_discrete(limits=names(pvca.means), labels=c(&quot;Sex&quot;, &quot;Pheno center&quot;, &quot;Residual&quot;)) +
  theme(axis.text.x = element_text(angle=0, vjust=1, hjust=.5))

pvca.plot.nobatch &lt;- pvca.plot.nobatch +
  labs(y=&quot;&quot;, title=&quot;&quot;) +
  scale_x_discrete(limits=names(pvca.means.nobatch), labels=c(&quot;Sex&quot;, &quot;Pheno center&quot;, &quot;Residual&quot;)) +
  theme(axis.text.x = element_text(angle=0, vjust=1, hjust=.5))

pvca.plots &lt;- ggarrange(pvca.plot, pvca.plot.nobatch, labels = &quot;AUTO&quot;)

png(file=&quot;docs/figure/figures.Rmd/pvca_BC_supp_v16.png&quot;, width=700, height=400)
pvca.plots
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="computing-phenotypic-correlations" class="section level3">
<h3>Computing phenotypic correlations</h3>
<p>We compute the phenotype correlations using different methods and
compare them.</p>
<pre class="r"><code>BC.cor.rank &lt;- cor(BC.mat.rank[,-1:-3], use=&quot;pairwise.complete.obs&quot;) # pearson correlation coefficient
BC.cor &lt;- cor(BC.mat[,-1:-3], use=&quot;pairwise.complete.obs&quot;, method=&quot;spearman&quot;) # spearman
BC.cor.combat &lt;- cor(t(combat_komp), use=&quot;pairwise.complete.obs&quot;)
pheno.list &lt;- rownames(BC.cor)

ht1 = Heatmap(BC.cor, show_column_names = F, row_names_gp = gpar(fontsize = 9), name=&quot;Spearm. Corr.&quot;)
draw(ht1)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-1.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ht2 = Heatmap(BC.cor.rank, show_column_names = F, row_names_gp = gpar(fontsize = 9), name=&quot;Corr. RankZ&quot;)
draw(ht2)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-2.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-2.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ht3 = Heatmap(BC.cor.combat, show_column_names = F, row_names_gp = gpar(fontsize = 9), name=&quot;Corr. ComBat&quot;)
draw(ht3)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-3">
Past versions of unnamed-chunk-4-3.png
</button>
</p>
<div id="fig-unnamed-chunk-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-3.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-4-3.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="preparation-of-impc-summary-statistics-data"
class="section level2">
<h2>Preparation of IMPC summary statistics data</h2>
<div id="loading-body-composition-summary-stat-impcv16"
class="section level3">
<h3>Loading Body Composition summary stat (IMPCv16)</h3>
<pre class="r"><code>BC.stat &lt;- readRDS(&quot;data/BC.stat.v16.rds&quot;)
dim(BC.stat)</code></pre>
<pre><code>[1] 56743     8</code></pre>
<pre class="r"><code>table(BC.stat$parameter_name, BC.stat$procedure_name)</code></pre>
<pre><code>                                        
                                           BC
  BMC/Body weight                        6943
  Body length                            6121
  Bone Area                              6945
  Bone Mineral Content (excluding skull) 5825
  Bone Mineral Density (excluding skull) 6945
  Fat mass                               2884
  Fat/Body weight                        7026
  Lean mass                              7028
  Lean/Body weight                       7026</code></pre>
<pre class="r"><code>length(unique(BC.stat$marker_symbol)) #6145</code></pre>
<pre><code>[1] 6145</code></pre>
<pre class="r"><code>length(unique(BC.stat$allele_symbol)) #6313</code></pre>
<pre><code>[1] 6313</code></pre>
<pre class="r"><code>length(unique(BC.stat$proc_param_name)) #9, number of phenotypes in association statistics data set</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>length(unique(BC.data$proc_param_name)) #10, number of phenotypes in final control data</code></pre>
<pre><code>[1] 10</code></pre>
<pre class="r"><code>pheno.list.stat &lt;- unique(BC.stat$proc_param_name)
pheno.list.ctrl &lt;- unique(BC.data$proc_param_name)
sum(pheno.list.stat %in% pheno.list.ctrl)</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>sum(pheno.list.ctrl %in% pheno.list.stat)</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code># Identifying common phenotypes between statistics and control data
common.pheno.list &lt;- sort(intersect(pheno.list.ctrl, pheno.list.stat))
length(common.pheno.list) # 9 - each data set had one phenotype not present in the other</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code># Filtering summary statistics to contain only common phenotypes
dim(BC.stat)</code></pre>
<pre><code>[1] 56743     8</code></pre>
<pre class="r"><code>BC.stat &lt;- BC.stat %&gt;% filter(proc_param_name %in% common.pheno.list)
dim(BC.stat)</code></pre>
<pre><code>[1] 56743     8</code></pre>
<pre class="r"><code>length(unique(BC.stat$proc_param_name))</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code># Exclude &#39;Fat mass&#39; due to limited gene availability
# only 2884 genes available in Fat mass
BC.stat &lt;- BC.stat %&gt;% filter(parameter_name != &quot;Fat mass&quot;) 
dim(BC.stat)</code></pre>
<pre><code>[1] 53859     8</code></pre>
<pre class="r"><code>length(unique(BC.stat$proc_param_name))</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>table(BC.stat$parameter_name, BC.stat$procedure_name)</code></pre>
<pre><code>                                        
                                           BC
  BMC/Body weight                        6943
  Body length                            6121
  Bone Area                              6945
  Bone Mineral Content (excluding skull) 5825
  Bone Mineral Density (excluding skull) 6945
  Fat/Body weight                        7026
  Lean mass                              7028
  Lean/Body weight                       7026</code></pre>
<pre class="r"><code>common.pheno.list &lt;- common.pheno.list[-6] 
common.pheno.list</code></pre>
<pre><code>[1] &quot;BC_BMC/Body weight&quot;                       
[2] &quot;BC_Body length&quot;                           
[3] &quot;BC_Bone Area&quot;                             
[4] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[5] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[6] &quot;BC_Fat/Body weight&quot;                       
[7] &quot;BC_Lean mass&quot;                             
[8] &quot;BC_Lean/Body weight&quot;                      </code></pre>
</div>
<div id="visualizing-gene-phenotype-pair-duplicates"
class="section level3">
<h3>Visualizing gene-phenotype pair duplicates</h3>
<pre class="r"><code>mtest &lt;- table(BC.stat$proc_param_name, BC.stat$marker_symbol)
mtest &lt;-as.data.frame.matrix(mtest)
nmax &lt;-max(mtest)
col_fun = colorRamp2(c(0, nmax), c(&quot;white&quot;, &quot;red&quot;))
col_fun(seq(0, nmax))</code></pre>
<pre><code> [1] &quot;#FFFFFFFF&quot; &quot;#FFF1ECFF&quot; &quot;#FFE4DAFF&quot; &quot;#FFD6C8FF&quot; &quot;#FFC8B6FF&quot; &quot;#FFBAA4FF&quot;
 [7] &quot;#FFAC93FF&quot; &quot;#FF9E81FF&quot; &quot;#FF8F70FF&quot; &quot;#FF805FFF&quot; &quot;#FF704FFF&quot; &quot;#FF5F3EFF&quot;
[13] &quot;#FF4B2DFF&quot; &quot;#FF331AFF&quot; &quot;#FF0000FF&quot;</code></pre>
<pre class="r"><code>ht = Heatmap(as.matrix(mtest), cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, col = col_fun,
             row_names_gp = gpar(fontsize = 8), name=&quot;Count&quot;)
draw(ht)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/gene_pheno_dup-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gene_pheno_dup-1">
Past versions of gene_pheno_dup-1.png
</button>
</p>
<div id="fig-gene_pheno_dup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/gene_pheno_dup-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="consolidating-muliple-z-scores-of-a-gene-phenotype-pair-using-stouffers-method"
class="section level3">
<h3>Consolidating muliple z-scores of a gene-phenotype pair using
Stouffer’s Method</h3>
<pre class="r"><code>## sum(z-score)/sqrt(# of zscore)
sumz &lt;- function(z){ sum(z)/sqrt(length(z)) }
BC.z = BC.stat %&gt;%
  dplyr::select(marker_symbol, proc_param_name, z_score) %&gt;%
  na.omit() %&gt;%
  group_by(marker_symbol, proc_param_name) %&gt;% 
  summarize(zscore = sumz(z_score)) ## combine z-scores
dim(BC.z)</code></pre>
<pre><code>[1] 46767     3</code></pre>
</div>
<div id="generating-z-score-matrix-reformatting" class="section level3">
<h3>Generating Z-score matrix (reformatting)</h3>
<pre class="r"><code># Function to convert NaN to NA
nan2na &lt;- function(df){ 
  out &lt;- data.frame(sapply(df, function(x) ifelse(is.nan(x), NA, x))) 
  colnames(out) &lt;- colnames(df)
  out
}

# Converting the long format of z-scores to a wide format matrix
BC.zmat = dcast(BC.z, marker_symbol ~ proc_param_name, value.var = &quot;zscore&quot;, 
             fun.aggregate = mean) %&gt;% tibble::column_to_rownames(var=&quot;marker_symbol&quot;)
BC.zmat = nan2na(BC.zmat) #convert nan to na

dim(BC.zmat)</code></pre>
<pre><code>[1] 6145    8</code></pre>
</div>
<div id="visualization-of-phenotype-gene-coverage"
class="section level3">
<h3>Visualization of Phenotype-Gene Coverage</h3>
<p>The heatmap illustrates tested (red) and untested (white)
gene-phenotype pairs.</p>
<pre class="r"><code># Generate a matrix indicating where z-scores are present
id.mat &lt;- 1*(!is.na(BC.zmat)) # multiply 1 to make this matrix numeric
nrow(as.data.frame(colSums(id.mat)))</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>dim(id.mat)</code></pre>
<pre><code>[1] 6145    8</code></pre>
<pre class="r"><code>ht = Heatmap(t(id.mat), 
             cluster_rows = T, clustering_distance_rows =&quot;binary&quot;,
             cluster_columns = T, clustering_distance_columns = &quot;binary&quot;,
             show_row_dend = F, show_column_dend = F,  # do not show dendrogram
             show_column_names = F, col = c(&quot;white&quot;,&quot;red&quot;),
             row_names_gp = gpar(fontsize = 10), name=&quot;Missing&quot;)
draw(ht)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-6-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-6-1.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-6-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="distribution-of-z-scores-across-phenotypes"
class="section level3">
<h3>Distribution of Z-Scores Across Phenotypes</h3>
<p>The histogram presents the distribution of association Z-scores for
each phenotype.</p>
<pre class="r"><code>ggplot(melt(BC.zmat), aes(x=value)) + 
  geom_histogram() + 
  facet_wrap(~variable, scales=&quot;free&quot;, ncol=5)+
  theme(strip.text.x = element_text(size = 6))</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-7-1.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-7-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="estimation-of-genetic-correlation-matrix-using-z-scores"
class="section level3">
<h3>Estimation of Genetic Correlation Matrix Using Z-Scores</h3>
<p>Here, we estimate the genetic correlations between phenotypes
utilizing the association Z-score matrix.</p>
<pre class="r"><code># Select common phenotypes
BC.zmat &lt;- BC.zmat[,common.pheno.list]
dim(BC.zmat)</code></pre>
<pre><code>[1] 6145    8</code></pre>
<pre class="r"><code># Compute genetic correlations
BC.zcor = cor(BC.zmat, use=&quot;pairwise.complete.obs&quot;)

# Generate heatmap of the correlation matrix
ht = Heatmap(BC.zcor, cluster_rows = T, cluster_columns = T, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 10),
             name=&quot;Genetic Corr (Z-score)&quot;
             )
draw(ht)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-8-1.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-8-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="comparison-of-phenotypic-correlation-and-genetic-correlation-among-phenotypes"
class="section level3">
<h3>Comparison of Phenotypic Correlation and Genetic Correlation Among
Phenotypes</h3>
<p>We will compare the correlation matrix obtained from control mice
phenotype data and the genetic correlation matrix estimated using
association Z-scores. As depicted below, both correlation heatmaps show
similar correlation patterns.</p>
<pre class="r"><code>BC.cor.rank.fig &lt;- BC.cor.rank[common.pheno.list,common.pheno.list]
BC.cor.fig &lt;- BC.cor[common.pheno.list,common.pheno.list]
BC.cor.combat.fig &lt;- BC.cor.combat[common.pheno.list, common.pheno.list]
BC.zcor.fig &lt;- BC.zcor


ht = Heatmap(BC.cor.rank.fig, cluster_rows = TRUE, cluster_columns = TRUE, show_column_names = F, #col = col_fun,
              show_row_dend = F, show_column_dend = F,  # do not show dendrogram
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (RankZ, Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
pheno.order &lt;- row_order(ht)
#draw(ht)

BC.cor.rank.fig &lt;- BC.cor.rank.fig[pheno.order,pheno.order]
ht1 = Heatmap(BC.cor.rank.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
              show_row_dend = F, show_column_dend = F,  # do not show dendrogram
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (RankZ, Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
BC.cor.fig &lt;- BC.cor.fig[pheno.order,pheno.order]  
ht2 = Heatmap(BC.cor.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (Spearman)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
BC.cor.combat.fig &lt;- BC.cor.combat.fig[pheno.order,pheno.order]  
ht3 = Heatmap(BC.cor.combat.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (Combat, Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
BC.zcor.fig &lt;- BC.zcor.fig[pheno.order,pheno.order]
ht4 = Heatmap(BC.zcor.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Genetic Corr  (Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;
             )
draw(ht1+ht2+ht3+ht4)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/comp_pheno_corr_gene_corr-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-comp_pheno_corr_gene_corr-1">
Past versions of comp_pheno_corr_gene_corr-1.png
</button>
</p>
<div id="fig-comp_pheno_corr_gene_corr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/comp_pheno_corr_gene_corr-1.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/comp_pheno_corr_gene_corr-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(file=&quot;docs/figure/figures.Rmd/cors_BC_v16.png&quot;, width=800, height=250)
draw(ht1+ht2+ht3+ht4)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code># Supplementary Figure 2
BC.cor.combat.fig2 &lt;- BC.cor.combat.fig
BC.zcor.fig2 &lt;- BC.zcor.fig
rownames(BC.cor.combat.fig2) &lt;- substring(rownames(BC.cor.combat.fig2), 4)
rownames(BC.zcor.fig2) &lt;- substring(rownames(BC.zcor.fig), 4)
rownames(BC.cor.combat.fig2)[c(6,7)] = c(&quot;Bone Mineral Content&quot;, &quot;Bone Mineral Density&quot;)
rownames(BC.zcor.fig2)[c(6,7)] = c(&quot;Bone Mineral Content&quot;, &quot;Bone Mineral Density&quot;)

ht3 = Heatmap(BC.cor.combat.fig2, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 12), column_title=&quot;Phenotypic Correlation&quot;, column_title_gp = gpar(fontsize = 15),
             name=&quot;Corr&quot;,
             heatmap_legend_param = list(title_gp=gpar(fontsize=15), labels_gp=gpar(fontsize=12),
                                         legend_height=unit(50, &quot;mm&quot;)))
ht4 = Heatmap(BC.zcor.fig2, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 12), column_title=&quot;Genetic Correlation&quot;, column_title_gp = gpar(fontsize = 15),
             name=&quot;Corr&quot;)

png(file=&quot;docs/figure/figures.Rmd/cors_BC_supp_v16.png&quot;, width=625, height=250)
draw(ht4+ht3)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>pdf(file=&quot;docs/figure/figures.Rmd/cors_BC_supp_v16.pdf&quot;, width=12, height=4.9)
draw(ht4+ht3)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div
id="correlation-analysis-between-genetic-correlation-matrices-using-mantels-test"
class="section level3">
<h3>Correlation Analysis Between Genetic Correlation Matrices Using
Mantel’s Test</h3>
<p>To evaluate the correlation between different genetic correlation
matrices, we apply Mantel’s test, which measures the correlation between
two distance matrices.</p>
<pre class="r"><code>####################
# Use Mantel test 
# https://stats.idre.ucla.edu/r/faq/how-can-i-perform-a-mantel-test-in-r/
# install.packages(&quot;ade4&quot;)
library(ade4)

# Function to extract upper triangular elements of a matrix
to.upper&lt;-function(X) X[upper.tri(X,diag=FALSE)]

a1 &lt;- to.upper(BC.cor.fig)
a2 &lt;- to.upper(BC.cor.rank.fig)
a3 &lt;- to.upper(BC.cor.combat.fig)
a4 &lt;- to.upper(BC.zcor.fig)

plot(a4, a1)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-9-1.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(a4, a2)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-9-2.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(a4, a3)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-3">
Past versions of unnamed-chunk-9-3.png
</button>
</p>
<div id="fig-unnamed-chunk-9-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/9b680ec50e9cdfae2fbbd653d60788ae4ff91a61/docs/figure/kompute_test_BC_v16.Rmd/unnamed-chunk-9-3.png" target="_blank">9b680ec</a>
</td>
<td>
statsleelab
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>mantel.rtest(as.dist(1-BC.cor.fig), as.dist(1-BC.zcor.fig), nrepet = 9999) #nrepet = number of permutations</code></pre>
<pre><code>Monte-Carlo test
Call: mantelnoneuclid(m1 = m1, m2 = m2, nrepet = nrepet)

Observation: 0.5396627 

Based on 9999 replicates
Simulated p-value: 8e-04 
Alternative hypothesis: greater 

    Std.Obs Expectation    Variance 
 2.57683170  0.00431246  0.04316223 </code></pre>
<pre class="r"><code>mantel.rtest(as.dist(1-BC.cor.rank.fig), as.dist(1-BC.zcor.fig), nrepet = 9999)</code></pre>
<pre><code>Monte-Carlo test
Call: mantelnoneuclid(m1 = m1, m2 = m2, nrepet = nrepet)

Observation: 0.5383613 

Based on 9999 replicates
Simulated p-value: 0.0013 
Alternative hypothesis: greater 

     Std.Obs  Expectation     Variance 
2.4666013697 0.0005465111 0.0475409136 </code></pre>
<pre class="r"><code>mantel.rtest(as.dist(1-BC.cor.combat.fig), as.dist(1-BC.zcor.fig), nrepet = 9999)</code></pre>
<pre><code>Monte-Carlo test
Call: mantelnoneuclid(m1 = m1, m2 = m2, nrepet = nrepet)

Observation: 0.6543789 

Based on 9999 replicates
Simulated p-value: 2e-04 
Alternative hypothesis: greater 

     Std.Obs  Expectation     Variance 
 2.608321834 -0.004513287  0.063812597 </code></pre>
</div>
</div>
<div id="evaluating-the-kompute-imputation-algorithm"
class="section level2">
<h2>Evaluating the KOMPUTE Imputation Algorithm</h2>
<div id="initializing-the-kompute-package" class="section level3">
<h3>Initializing the KOMPUTE Package</h3>
<pre class="r"><code># Check if KOMPUTE is installed, if not, install it from GitHub using devtools
if(!&quot;kompute&quot; %in% rownames(installed.packages())){
  library(devtools)
  devtools::install_github(&quot;dleelab/kompute&quot;)
}
library(kompute)</code></pre>
</div>
<div
id="simulation-study---comparison-of-imputed-vs-measured-z-score-values"
class="section level3">
<h3>Simulation study - Comparison of imputed vs measured z-score
values</h3>
<p>In this section, we conduct a simulation study to compare the
performance of the KOMPUTE method with the measured gene-phenotype
association z-scores. We randomly select some of these measured
z-scores, mask them, and then use the KOMPUTE method to impute them. We
then compare the imputed z-scores with the measured ones.</p>
<pre class="r"><code>zmat &lt;-t(BC.zmat) 
dim(zmat)</code></pre>
<pre><code>[1]    8 6145</code></pre>
<pre class="r"><code># filter genes with less than 1 missing data point (na)
zmat0 &lt;- is.na(zmat)
num.na&lt;-colSums(zmat0)
summary(num.na)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.3894  1.0000  7.0000 </code></pre>
<pre class="r"><code>dim(zmat)</code></pre>
<pre><code>[1]    8 6145</code></pre>
<pre class="r"><code>dim(zmat[,num.na&lt;1])</code></pre>
<pre><code>[1]    8 4350</code></pre>
<pre class="r"><code>dim(zmat[,num.na&lt;5])</code></pre>
<pre><code>[1]    8 6025</code></pre>
<pre class="r"><code>dim(zmat[,num.na&lt;10])</code></pre>
<pre><code>[1]    8 6145</code></pre>
<pre class="r"><code># filter genes with less than 1 missing data point (na)
zmat &lt;- zmat[,num.na&lt;1]
dim(zmat)</code></pre>
<pre><code>[1]    8 4350</code></pre>
<pre class="r"><code># Set correlation method for phenotypes
#pheno.cor &lt;- BC.cor.fig
#pheno.cor &lt;- BC.cor.rank.fig
pheno.cor &lt;- BC.cor.combat.fig
#pheno.cor &lt;- BC.zcor.fig


zmat &lt;- zmat[rownames(pheno.cor),,drop=FALSE]
rownames(zmat)</code></pre>
<pre><code>[1] &quot;BC_Bone Area&quot;                             
[2] &quot;BC_Lean mass&quot;                             
[3] &quot;BC_Body length&quot;                           
[4] &quot;BC_Fat/Body weight&quot;                       
[5] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[6] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[7] &quot;BC_BMC/Body weight&quot;                       
[8] &quot;BC_Lean/Body weight&quot;                      </code></pre>
<pre class="r"><code>rownames(pheno.cor)</code></pre>
<pre><code>[1] &quot;BC_Bone Area&quot;                             
[2] &quot;BC_Lean mass&quot;                             
[3] &quot;BC_Body length&quot;                           
[4] &quot;BC_Fat/Body weight&quot;                       
[5] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[6] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[7] &quot;BC_BMC/Body weight&quot;                       
[8] &quot;BC_Lean/Body weight&quot;                      </code></pre>
<pre class="r"><code>colnames(pheno.cor)</code></pre>
<pre><code>[1] &quot;BC_Bone Area&quot;                             
[2] &quot;BC_Lean mass&quot;                             
[3] &quot;BC_Body length&quot;                           
[4] &quot;BC_Fat/Body weight&quot;                       
[5] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[6] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[7] &quot;BC_BMC/Body weight&quot;                       
[8] &quot;BC_Lean/Body weight&quot;                      </code></pre>
<pre class="r"><code>npheno &lt;- nrow(zmat)

# calculate the percentage of missing Z-scores in the original data 
100*sum(is.na(zmat))/(nrow(zmat)*ncol(zmat)) # 0%</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>nimp &lt;- 1000 # # of missing/imputed Z-scores
set.seed(1234)

## find index of all measured zscores
all.i &lt;- 1:(nrow(zmat)*ncol(zmat))
measured &lt;- as.vector(!is.na(as.matrix(zmat)))
measured.i &lt;- all.i[measured]

## mask 2000 measured z-scores
mask.i &lt;- sort(sample(measured.i, nimp))
org.z = as.matrix(zmat)[mask.i]
zvec &lt;- as.vector(as.matrix(zmat))
zvec[mask.i] &lt;- NA
zmat.imp &lt;- matrix(zvec, nrow=npheno)
rownames(zmat.imp) &lt;- rownames(zmat)</code></pre>
</div>
<div id="run-kompute-method" class="section level3">
<h3>Run KOMPUTE method</h3>
<pre class="r"><code>kompute.res &lt;- kompute(t(zmat.imp), pheno.cor, 0.01)

# Compare measured vs imputed z-scores
length(org.z)</code></pre>
<pre><code>[1] 1000</code></pre>
<pre class="r"><code>imp.z &lt;- as.matrix(t(kompute.res$zmat))[mask.i]
imp.info &lt;- as.matrix(t(kompute.res$infomat))[mask.i]  
plot(imp.z, org.z)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-KOMPUTE_res-1">
Past versions of KOMPUTE_res-1.png
</button>
</p>
<div id="fig-KOMPUTE_res-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/989feb60f6281fb4eaf2cfcfce252144832ee843/docs/figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-1.png" target="_blank">989feb6</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-1.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Create a dataframe with the original and imputed z-scores and the information of imputed z-scores
imp &lt;- data.frame(org.z=org.z, imp.z=imp.z, info=imp.info)
dim(imp)</code></pre>
<pre><code>[1] 1000    3</code></pre>
<pre class="r"><code>imp &lt;- imp[complete.cases(imp),]
imp &lt;- subset(imp, info&gt;=0 &amp; info &lt;= 1)
dim(imp)</code></pre>
<pre><code>[1] 1000    3</code></pre>
<pre class="r"><code>cor.val &lt;- round(cor(imp$imp.z, imp$org.z), digits=3)
cor.val</code></pre>
<pre><code>[1] 0.793</code></pre>
<pre class="r"><code>plot(imp$imp.z, imp$org.z)</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-KOMPUTE_res-2">
Past versions of KOMPUTE_res-2.png
</button>
</p>
<div id="fig-KOMPUTE_res-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/989feb60f6281fb4eaf2cfcfce252144832ee843/docs/figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-2.png" target="_blank">989feb6</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-2.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Set a cutoff for information content and filter the data accordingly
info.cutoff &lt;- 0.8
imp.sub &lt;- subset(imp, info&gt;info.cutoff)
dim(imp.sub)</code></pre>
<pre><code>[1] 583   3</code></pre>
<pre class="r"><code>summary(imp.sub$imp.z)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
-9.83798 -0.78293  0.09257  0.03896  0.96072  6.43130 </code></pre>
<pre class="r"><code>summary(imp.sub$info)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.8081  0.8478  0.8517  0.8648  0.8748  0.9297 </code></pre>
<pre class="r"><code>cor.val &lt;- round(cor(imp.sub$imp.z, imp.sub$org.z), digits=3)
cor.val</code></pre>
<pre><code>[1] 0.841</code></pre>
<pre class="r"><code>g &lt;- ggplot(imp.sub, aes(x=imp.z, y=org.z, col=info)) +
    geom_point() +
    labs(title=paste0(&quot;IMPC BC Data, Info&gt;&quot;, info.cutoff, &quot;, Cor=&quot;,cor.val),
      x=&quot;Imputed Z-scores&quot;, y = &quot;Measured Z-scores&quot;, col=&quot;Info&quot;) +
    theme_minimal()
g</code></pre>
<p><img src="figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-3.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-KOMPUTE_res-3">
Past versions of KOMPUTE_res-3.png
</button>
</p>
<div id="fig-KOMPUTE_res-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/989feb60f6281fb4eaf2cfcfce252144832ee843/docs/figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-3.png" target="_blank">989feb6</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/statsleelab/komputeExamples/blob/7685a0969564d21706311b2c9c3aa70da7ebd2d4/docs/figure/kompute_test_BC_v16.Rmd/KOMPUTE_res-3.png" target="_blank">7685a09</a>
</td>
<td>
statsleelab
</td>
<td>
2023-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># save plot
png(file=&quot;docs/figure/figures.Rmd/sim_results_BC_v16.png&quot;, width=600, height=350)
g
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code># Part 1 of Figure 2
fig2.1 &lt;- ggplot(imp.sub, aes(x=imp.z, y=org.z, col=info)) +
  geom_point() +
  labs(title=&quot;Body Composition&quot;,
       x=&quot;Imputed Z-scores&quot;, y = &quot;Measured Z-scores&quot;, col=&quot;Info&quot;) +
  scale_x_continuous(limits=c(-9,9), breaks=c(seq(-9,9,3)), minor_breaks = NULL) +
  scale_y_continuous(limits=c(-9,9), breaks=c(seq(-9,9,3))) +
  scale_color_gradient(limits=c(0.8,1), low=&quot;#98cdf9&quot;, high=&quot;#084b82&quot;) +
  theme_bw() +
  theme(legend.position=&quot;none&quot;, plot.title=element_text(hjust=0.5))
save(fig2.1, file=&quot;docs/figure/figures.Rmd/sim_BC_v16.rdata&quot;)</code></pre>
</div>
<div id="save-z-score-matrix-and-phenotype-correlation-matrix"
class="section level3">
<h3>Save z-score matrix and phenotype correlation matrix</h3>
<pre class="r"><code>plist &lt;- sort(colnames(BC.zmat))
BC_Zscore_Mat &lt;- as.matrix(BC.zmat[,plist])
save(BC_Zscore_Mat, file = &quot;data/BC.zmat.v16.RData&quot;)

BC_Pheno_Cor &lt;- pheno.cor[plist,plist]
save(BC_Pheno_Cor, file = &quot;data/BC.pheno.cor.v16.RData&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] kompute_0.1.0         ade4_1.7-20           sva_3.44.0           
 [4] BiocParallel_1.30.3   genefilter_1.78.0     mgcv_1.8-40          
 [7] nlme_3.1-158          lme4_1.1-31           Matrix_1.5-1         
[10] RNOmni_1.0.1          ggpubr_0.5.0          patchwork_1.1.2      
[13] ComplexHeatmap_2.12.1 circlize_0.4.15       RColorBrewer_1.1-3   
[16] tidyr_1.2.0           ggplot2_3.4.1         reshape2_1.4.4       
[19] dplyr_1.0.9           data.table_1.14.2     workflowr_1.7.0.1    

loaded via a namespace (and not attached):
  [1] minqa_1.2.5            colorspace_2.1-0       ggsignif_0.6.4        
  [4] rjson_0.2.21           rprojroot_2.0.3        XVector_0.36.0        
  [7] GlobalOptions_0.1.2    fs_1.5.2               clue_0.3-62           
 [10] rstudioapi_0.13        farver_2.1.1           bit64_4.0.5           
 [13] AnnotationDbi_1.58.0   fansi_1.0.4            codetools_0.2-18      
 [16] splines_4.2.1          doParallel_1.0.17      cachem_1.0.6          
 [19] knitr_1.39             jsonlite_1.8.0         nloptr_2.0.3          
 [22] broom_1.0.1            annotate_1.74.0        cluster_2.1.3         
 [25] png_0.1-8              compiler_4.2.1         httr_1.4.3            
 [28] backports_1.4.1        assertthat_0.2.1       fastmap_1.1.0         
 [31] limma_3.52.4           cli_3.6.0              later_1.3.0           
 [34] htmltools_0.5.3        tools_4.2.1            GenomeInfoDbData_1.2.8
 [37] gtable_0.3.1           glue_1.6.2             Rcpp_1.0.10           
 [40] carData_3.0-5          Biobase_2.56.0         jquerylib_0.1.4       
 [43] vctrs_0.5.2            Biostrings_2.64.0      iterators_1.0.14      
 [46] xfun_0.31              stringr_1.4.0          ps_1.7.1              
 [49] lifecycle_1.0.3        rstatix_0.7.1          XML_3.99-0.10         
 [52] edgeR_3.38.4           zlibbioc_1.42.0        getPass_0.2-2         
 [55] MASS_7.3-58.1          scales_1.2.1           promises_1.2.0.1      
 [58] parallel_4.2.1         yaml_2.3.5             memoise_2.0.1         
 [61] sass_0.4.2             stringi_1.7.8          RSQLite_2.2.15        
 [64] highr_0.9              S4Vectors_0.34.0       foreach_1.5.2         
 [67] BiocGenerics_0.42.0    boot_1.3-28            shape_1.4.6           
 [70] GenomeInfoDb_1.32.3    rlang_1.0.6            pkgconfig_2.0.3       
 [73] matrixStats_0.62.0     bitops_1.0-7           evaluate_0.16         
 [76] lattice_0.20-45        purrr_0.3.4            labeling_0.4.2        
 [79] cowplot_1.1.1          bit_4.0.4              processx_3.7.0        
 [82] tidyselect_1.2.0       plyr_1.8.7             magrittr_2.0.3        
 [85] R6_2.5.1               IRanges_2.30.0         generics_0.1.3        
 [88] DBI_1.1.3              pillar_1.8.1           whisker_0.4           
 [91] withr_2.5.0            survival_3.3-1         KEGGREST_1.36.3       
 [94] abind_1.4-5            RCurl_1.98-1.8         tibble_3.1.8          
 [97] crayon_1.5.1           car_3.1-1              utf8_1.2.3            
[100] rmarkdown_2.14         GetoptLong_1.0.5       locfit_1.5-9.6        
[103] blob_1.2.3             callr_3.7.1            git2r_0.30.1          
[106] digest_0.6.29          xtable_1.8-4           httpuv_1.6.5          
[109] stats4_4.2.1           munsell_0.5.0          bslib_0.4.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
